core/variable_operations.cpp

```diff
@@ -206,12 +206,12 @@ Variable sum(const VariableConstProxy &var, const Dim dim) {
 
 Variable sum(const VariableConstProxy &var, const Dim dim,
              const MasksConstProxy &masks) {
-  if (masks.empty()) {
-    return sum(var, dim);
-  } else {
-    const auto mask_union = masks_merge(masks);
-    return sum(var * ~mask_union, dim);
+  if (!masks.empty()) {
+    const auto mask_union = masks_merge(masks, dim);
+    if (mask_union.dims().contains(dim))
```

Comment:
Do we need this check if `masks_merge` filters all masks with unrelated dimensions now?

Reply:
Yes, because `mask_union` will be 0-D in that case and we want to skip the multiplication?

Reply:
I see. That makes sense